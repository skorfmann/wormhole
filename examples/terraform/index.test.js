const { TestDriver } = require('./test-helper')

jest.setTimeout(120000)

describe("full integration test synth", () => {
  let driver;

  beforeAll(() => {
    driver = new TestDriver(__dirname)
    driver.setupTypescriptProject()
  });

  afterAll(() => {
    driver.reset()
  })

  test("synth generates JSON", async () => {
    driver.synth()
    expect(driver.synthesizedStack()).toMatchSnapshot()
  })
})