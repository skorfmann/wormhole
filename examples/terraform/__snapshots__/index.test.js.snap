// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full integration test synth synth generates JSON 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"hello-terra\\"
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"aws\\": {
        \\"version\\": \\"~> 2.0\\",
        \\"source\\": \\"aws\\"
      }
    },
    \\"backend\\": {
      \\"remote\\": {
        \\"organization\\": \\"test\\",
        \\"workspaces\\": {
          \\"name\\": \\"test\\"
        }
      }
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"region\\": \\"eu-central-1\\",
        \\"ignore_tags\\": [
          {
            \\"keys\\": [
              \\"foo\\"
            ]
          }
        ]
      }
    ]
  },
  \\"resource\\": {
    \\"aws_sns_topic\\": {
      \\"Topic\\": {
        \\"display_name\\": \\"topic\\",
        \\"provider\\": \\"aws\\",
        \\"lifecycle\\": {
          \\"create_before_destroy\\": true
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"hello-terra/Topic\\",
            \\"uniqueId\\": \\"Topic\\"
          }
        }
      }
    }
  },
  \\"output\\": {
    \\"snstopicarn\\": {
      \\"value\\": \\"\${aws_sns_topic.Topic.arn}\\"
    }
  }
}"
`;
